testsuite/MDAnalysisTests/analysis/test_msd.py

```diff
@@ -0,0 +1,219 @@
+    m_fft.run()
+    timeseries_fft = m_fft.timeseries
+    assert_almost_equal(timeseries_simple, timeseries_fft, decimal=4)
+
+
+@pytest.mark.parametrize("dim", ['xyz', 'xy', 'xz', 'yz', 'x', 'y', 'z'])
+def test_fft_vs_simple_all_dims_per_particle(u, SELECTION, dim):
+    # check fft and simple give same result for each particle in each dimension
+    m_simple = MSD(u, SELECTION, msd_type=dim, fft=False)
+    m_simple.run()
+    per_particle_simple = m_simple.msds_by_particle
+    m_fft = MSD(u, SELECTION, msd_type=dim, fft=True)
+    m_fft.run()
+    per_particle_fft = m_fft.msds_by_particle
+    assert_almost_equal(per_particle_simple, per_particle_fft, decimal=4)
+
+
+@pytest.mark.parametrize("dim, dim_factor", \
+[('xyz', 3), ('xy', 2), ('xz', 2), ('yz', 2), ('x', 1), ('y', 1), ('z', 1)])
+def test_simple_step_traj_all_dims(step_traj, NSTEP, dim, dim_factor):
+    # testing the "simple" algorithm on constant velocity trajectory
+    # should fit the polynomial y=dim_factor*x**2
+    m_simple = MSD(step_traj, 'all', msd_type=dim, fft=False)
+    m_simple.run()
+    poly = characteristic_poly(NSTEP, dim_factor)
+    assert_almost_equal(m_simple.timeseries, poly, decimal=4)
+
+@pytest.mark.parametrize("dim, dim_factor", \
+[('xyz', 3), ('xy', 2), ('xz', 2), ('yz', 2), ('x', 1), ('y', 1), ('z', 1)])
+def test_simple_start_stop_step_all_dims(step_traj, NSTEP, dim, dim_factor):
+    # testing the "simple" algorithm on constant velocity trajectory
+    # test start stop step is working correctly
+    m_simple = MSD(step_traj, 'all', msd_type=dim, fft=False, start=10, \
+         stop=1000, step=10)
+    m_simple.run()
+    poly = characteristic_poly(NSTEP, dim_factor)
+    assert_almost_equal(m_simple.timeseries, poly[10:1000:10], decimal=4)
+
+
+@pytest.mark.parametrize("dim, dim_factor", \
+[('xyz', 3), ('xy', 2), ('xz', 2), ('yz', 2), ('x', 1), ('y', 1), ('z', 1)])
+def test_fft_step_traj_all_dims(step_traj, NSTEP, dim, dim_factor):
+    # testing the fft algorithm on constant velocity trajectory
+    # this should fit the polynomial y=dim_factor*x**2
+    # fft based tests require a slight decrease in expected prescision
+    # primarily due to roundoff in fft(ifft()) calls.
+    # relative accuracy expected to be around ~1e-12
+    m_simple = MSD(step_traj, 'all', msd_type=dim, fft=True)
+    m_simple.run()
+    poly = characteristic_poly(NSTEP, dim_factor)
+    # this was relaxed from decimal=4 for numpy=1.13 test
+    assert_almost_equal(m_simple.timeseries, poly, decimal=3)
+
+@pytest.mark.parametrize("dim, dim_factor", \
+[('xyz', 3), ('xy', 2), ('xz', 2), ('yz', 2), ('x', 1), ('y', 1), ('z', 1)])
+def test_fft_start_stop_step_all_dims(step_traj, NSTEP, dim, dim_factor):
+    # testing the fft algorithm on constant velocity trajectory
+    # test start stop step is working correctly
+    m_simple = MSD(step_traj, 'all', msd_type=dim, fft=True, start=10, \
+         stop=1000, step=10)
```

Comment:
The start/stop/step need to be passed to run():
```suggestion
    m_simple = MSD(step_traj, 'all', msd_type=dim, fft=True).run(
                                 start=10, stop=1000, step=10)
```